hardware_interface/include/hardware_interface/actuator_interface.hpp

```diff
@@ -202,8 +290,35 @@ class ActuatorInterface : public rclcpp_lifecycle::node_interfaces::LifecycleNod
    */
   void set_state(const rclcpp_lifecycle::State & new_state) { lifecycle_state_ = new_state; }
 
+  void set_state(const std::string & interface_name, const double & value)
+  {
+    actuator_states_.at(interface_name)->set_value(value);
+  }
+
+  double get_state(const std::string & interface_name) const
+  {
+    return actuator_states_.at(interface_name)->get_value();
+  }
+
+  void set_command(const std::string & interface_name, const double & value)
+  {
+    actuator_commands_.at(interface_name)->set_value(value);
+  }
+
+  double get_command(const std::string & interface_name) const
+  {
+    return actuator_commands_.at(interface_name)->get_value();
+  }
+
 protected:
   HardwareInfo info_;
+  std::map<std::string, InterfaceDescription> joint_state_interfaces_;
+  std::map<std::string, InterfaceDescription> joint_command_interfaces_;
+
+private:
+  std::map<std::string, std::shared_ptr<StateInterface>> actuator_states_;
+  std::map<std::string, std::shared_ptr<CommandInterface>> actuator_commands_;
```

Comment:
```suggestion
  std::unordered_map<std::string, InterfaceDescription> joint_state_interfaces_;
  std::unordered_map<std::string, InterfaceDescription> joint_command_interfaces_;

private:
  std::unordered_map<std::string, std::shared_ptr<StateInterface>> actuator_states_;
  std::unordered_map<std::string, std::shared_ptr<CommandInterface>> actuator_commands_;
```
use unordered_map instead od map

Reply:
ah yes good point